hy/contrib/hy_repr.hy

```diff
@@ -80,9 +80,9 @@
         (.rstrip (repr x) "L")
       (and (in t [float HyFloat]) (isnan x))
         "NaN"
-      (= x Inf)
+      (and (in t [float HyFloat]) (= x Inf))
         "Inf"
-      (= x -Inf)
+      (and (in t [float HyFloat]) (= x -Inf))
```

Comment:
Throughout this function, I opted for checking type equality rather than subclasses or instances so that if somebody makes a subclass, it won't get hy-repred to look exactly like an instance of its parent class.